### PR TITLE
Add file watcher for scheduler config file

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Address #1015 

Have manually tested it since it is difficult to add a test

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -86,19 +103,58 @@ func (pc *Scheduler) runOnce() {
 
 func (pc *Scheduler) loadSchedulerConf() {
 	var err error
+	pc.once.Do(func() {
```
Comment (k82cn) [inline pkg/scheduler/scheduler.go:114]: Why `once`, which make it call only once?

Comment (hzxuzhonghu) [reply pkg/scheduler/scheduler.go:114]: Yeah, this is the default configuration, we need load it only once, and if user provides a config file, then load the provided one. If the provoded one is invalid, fallback to the default one and log warning.

Comment (k82cn) [reply pkg/scheduler/scheduler.go:114]: Ok, that makes sense to me :)

**Pull request merged**